### PR TITLE
Weather App

### PR DESCRIPTION
**Pull request by haraphat01**
The project tests on the understanding of API, this is a weather app that is used to check the weather condition of any location, the user also has the opportunity to select Celsius or Fahrenheit.

Kindly check for your kind approval

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// const app = () => {
+//   const apiRequest = new XMLHttpRequest();
+//   const parent = document.getElementById('container');
+//   parent.addEventListener('click', (e) => {
+//     if (e.target.classList.contains('submit_button')) {
+//       e.preventDefault();
+//       const cityInput = document.getElementById('city');
+//       const chosenCity = cityInput.value;
```
Comment (Torres-ssf) [inline src/app.js]: - [ ] Since we are about to push this code to production, we should delete all commented code.

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi There @haraphat01 :hand: 
Great job on this milestone so far :clap: 
But we need to make some changes before moving on :smile: 

- [ ] The node_modules was added to the repository before adding it to .gitignore. We need to delete node_modules from the repository.

- [ ] As it said in the requirements, the application should change its appearance based on the received data.
Screenshot from 2020-10-28 11-53-00

- [ ] No scripts were added to the `package.json`. We need to add at least 2 scripts, one for development and another one for production. Both of them should specify in each mode Webpack should run:
```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
  },
``` 
Screenshot from 2020-10-28 12-09-23

- [ ] The error handling solution it's not showing anything for the user right now. We need to implement a different approach that shows a friendly message for the user, saying that the city name was not found (or something similar). You can use a hidden container with a message that is only visible when an error occurs in the fetch process.

**OPTIONAL**

- [ ] You can also add `webpack-dev-server` package to your dependencies and add an additional script:
```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```
This script will make it easier for users to test your application.

Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
#### Hello Team 👋
#### Status: Changes Required ♻️

Great job so far! 👏👋
Let's cover a few details together to make your project better 😊⏬

### Clean Repository 🍀

- [ ] Files not related to the project should not be committed and added to .gitignore (e.g. `.DS_Store`) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

💡 node_modules are still in the repository. Please make a quick research about "How to delete committed files from the repository". 😊

### Application is Crushing After Clicking "Check Weather"

- [ ] Please fix this. 👋😇

image

After that please don't forget to work on this too, as previous TSE mentioned:

- [ ]  `"As it said in the requirements, the application should change its appearance based on the received data."`

### Descriptive commits

- [ ] Commits are descriptive ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_commits.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

image

You can commit like this
- `Delete node_modules`
- `Fix application crush problem`
- `Add npm scripts`
- `Add instructions to readme`

This would be amazing 😊👏

### NPM scripts

- [ ] Necessary run common commands like `build`, `development` should be in `package.json` inside `scripts'. Here is a basic example ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

You can read this material to know more about this:
https://michael-kuehnel.de/tooling/2018/03/22/helpers-and-tips-for-npm-run-scripts.html#:~:text=You%20can%20easily%20run%20scripts,instead%20of%20pointing%20to%20node_modules%2F.


### README

- [ ] [OPTIONAL] Following readme setup guidelines is now working. There is no index.html in the root directory but you are pointing it out in your readme. You can improve this to help your users to see your great project 😊 You should tell them step-by-step what to do.

## That's it, you can submit another code review request after making these changes ♻️😇

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

**Comment by haraphat01** [comment]
I just reworked the TSE recommendation. The node modules have been deleted, the app crushing fixed

File: package.json [hunk 2]
```diff
@@ -6,10 +6,27 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1"
   },
```
Comment (christianotieno) [inline package.json]: ## Add scripts here to smoothly run and test the application
- [ ] You are required to include the required scripts to this `package.json`  file. We need to aim for at least 2 scripts, one for development and another one for production and both needs to specify in each mode Webpack should run: As in the example below 👇 

```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
  },
``` 

## Optional Change
- [ ] Also while at it, you are welcome to add the `webpack-dev-server` package to your dependencies and add an additional script. So in overall, we might have something that looks like this 👇 in the `package.json` file:

```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```

**Review by christianotieno — CHANGES_REQUESTED** [review]
### Changes Required ♻️ 
Hello @haraphat01 👋 
Good job working on the changes requested. we still have one small change to conduct before moving ahead below. 👇 

### Happy coding 💻 

#### Christian Otieno (TSE)

**Comment by haraphat01** [comment]
I added to my script file as it was instructed.

**Review by rOluochKe — APPROVED** [review]
Hello @haraphat01 
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**